### PR TITLE
Detach storage driver from materializations

### PR DESCRIPTION
**Pull request by csansoon**
## Describe your changes

A Latitude project is defined by the files the user configures (Views, Queries & latitude.json). In addition to those files, Latitude will create multiple files for caching queries and materialization that may take up space depending on the project queries and usage. Now, the user can configure where all of these internal files are saved in their `latitude.json` configuration file.

```json
{
  "name": "my-awesome-latitude-project",
  "version": "x.x.x",
  "storage": {
    "type": "disk",
    "path": "/custom/storage/path"
  }
}
```

By default, these files will be saved inside your project in `.latitude/app/static/.latitude/storage`, but you can change where this is saved.

https://github.com/latitude-dev/latitude/assets/57395395/3511eb8f-6e5f-4cbd-adae-6e416aa3d24c

> [!NOTE]
> Spoiler: In future PRs, you will be able to use custom S3 buckets for this storages.

Now, the materialization process is responsibility of the Source Manager. The Storage Driver only responsibility is the management of these temporal files, which are important for the performance of the app but may take up lots of space in the long term.

## Checklist before requesting a review
- [x] I have added thorough tests
- [ ] I have updated the documentation if necessary
- [x] I have added a human-readable description of the changes for the release notes
- [x] I have included a recorded video capture of the feature manually tested

### CODE REVIEW

File: packages/source_manager/package.json [hunk 1]
```diff
@@ -16,9 +16,12 @@
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
   "dependencies": {
+    "@aws-sdk/client-s3": "^3.592.0",
+    "@aws-sdk/lib-storage": "^3.596.0",
```
Comment (andresgutgon) [inline packages/source_manager/package.json:20]: Why source manager needs these dependencies?

Comment (csansoon) [reply packages/source_manager/package.json:20]: spoilers for my next PR

File: packages/storage_driver/package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    "@latitude-data/query_result": "workspace:^",
+    "@latitude-data/typescript": "workspace:*",
+    "@rollup/plugin-commonjs": "^25.0.7",
+    "@rollup/plugin-typescript": "^11.1.6",
+    "@types/mock-fs": "^4.13.4",
+    "@types/node": "^20.10.6",
+    "mock-fs": "^5.2.0",
+    "rollup": "^4.10.0",
+    "ts-node": "^10.9.1",
+    "tsx": "^4.15.2",
```
Comment (andresgutgon) [inline packages/storage_driver/package.json:33]: Review

File: packages/source_manager/package.json [hunk 3]
```diff
@@ -31,9 +32,13 @@
...
     "@types/mock-fs": "^4.13.4",
     "@types/node": "^20.10.6",
+    "@types/uuid": "^9.0.8",
     "mock-fs": "^5.2.0",
     "rollup": "^4.10.0",
-    "typescript": "^5.2.2",
+    "ts-node": "^10.9.1",
+    "tsx": "^4.15.2",
+    "typescript": "^5.4.3",
+    "uuid": "^9.0.1",
```
Comment (andresgutgon) [inline packages/source_manager/package.json:41]: Review

**Review by andresgutgon — APPROVED** [review]

File: packages/storage_driver/package.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "lint": "eslint . --max-warnings 0 --ext .ts src",
+    "tc": "tsc --noEmit",
+    "test:watch": "vitest",
+    "prettier": "prettier --write src/**/*.ts"
+  },
+  "type": "module",
+  "main": "dist/index.js",
+  "types": "dist/index.d.ts",
+  "dependencies": {
+    "dotenv": "^16.4.5",
```
Comment (andresgutgon) [inline packages/storage_driver/package.json:19]: Do we need dotenv here?

Comment (csansoon) [reply packages/storage_driver/package.json:19]: no

File: packages/storage_driver/package.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    "yaml": "^2.3.4"
```
Comment (andresgutgon) [inline packages/storage_driver/package.json:20]: What for?

File: packages/storage_driver/package.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  "devDependencies": {
+    "@latitude-data/eslint-config": "workspace:*",
+    "@latitude-data/query_result": "workspace:^",
```
Comment (andresgutgon) [inline packages/storage_driver/package.json:24]: Is this needed?

File: packages/storage_driver/package.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    "@latitude-data/typescript": "workspace:*",
+    "@rollup/plugin-commonjs": "^25.0.7",
```
Comment (andresgutgon) [inline packages/storage_driver/package.json:26]: Is this needed?

File: packages/storage_driver/package.json (hunk 6 above)
Comment (csansoon) [reply packages/storage_driver/package.json]: 😭

File: packages/storage_driver/package.json (hunk 7 above)
Comment (csansoon) [reply packages/storage_driver/package.json:21]: Yes! These are used by rollup :)

**Pull request merged**